File: app/src/main/java/org/tasks/time/DateTimeUtils.java
package org.tasks.time;

import android.annotation.SuppressLint;
import java.util.Date;
import org.tasks.BuildConfig;

public class DateTimeUtils {

  private static final SystemMillisProvider SYSTEM_MILLIS_PROVIDER = new SystemMillisProvider();
  private static volatile MillisProvider MILLIS_PROVIDER = SYSTEM_MILLIS_PROVIDER;

  public static long currentTimeMillis() {
    return MILLIS_PROVIDER.getMillis();
  }

  public static void setCurrentMillisFixed(long millis) {
    MILLIS_PROVIDER = new FixedMillisProvider(millis);
  }

  public static void setCurrentMillisSystem() {
    MILLIS_PROVIDER = SYSTEM_MILLIS_PROVIDER;
  }

  public static String printTimestamp(long timestamp) {
    return BuildConfig.DEBUG ? new Date(timestamp).toString() : Long.toString(timestamp);
  }

  @SuppressLint("DefaultLocale")
  public static String printDuration(long millis) {
    if (BuildConfig.DEBUG) {
      long seconds = millis / 1000;
      return String.format(
          "%dh %dm %ds", seconds / 3600L, (int) (seconds % 3600L / 60L), (int) (seconds % 60L));
    } else {
      return Long.toString(millis);
    }
  }
}

File: app/src/main/java/org/tasks/time/DateTimeUtils.kt
package org.tasks.time

import android.annotation.SuppressLint
import org.tasks.BuildConfig
import org.tasks.date.DateTimeUtils.toDateTime

object DateTimeUtils {
    @SuppressLint("DefaultLocale")
    fun printDuration(millis: Long): String = if (BuildConfig.DEBUG) {
        val seconds = millis / 1000
        String.format(
                "%dh %dm %ds", seconds / 3600L, (seconds % 3600L / 60L).toInt(), (seconds % 60L).toInt())
    } else millis.toString()

    fun Long.toDate(): net.fortuna.ical4j.model.Date? = this.toDateTime().toDate()
}
